packages/api/src/lib/prepareParams.js

```diff
@@ -83,68 +135,60 @@ module.exports = async (operation, body, metadata) => {
-          if (fs.existsSync(file)) {
-            conversions.push(
-              new Promise(resolve => {
-                resolve(datauri(file));
-              }).then(dataurl => {
-                // Doing this manually for now until when/if https://github.com/data-uri/datauri/pull/29 is accepted.
-                params.body[prop] = dataurl.replace(
-                  ';base64',
-                  `;name=${encodeURIComponent(path.basename(file))};base64`
-                );
-
-                return Promise.resolve(true);
-              })
-            );
+        const conversions = [];
+
+        // @todo add support for `type: array`, `oneOf` and `anyOf`
+        if (payloadJsonSchema.schema?.properties) {
+          Object.entries(payloadJsonSchema.schema?.properties)
+            .filter(([, schema]) => schema?.format === 'binary')
+            .filter(([prop]) => Object.keys(params.body).includes(prop))
+            .forEach(([prop]) => {
+              conversions.push(processFile(prop, params.body[prop]));
+            });
+        } else if (payloadJsonSchema.schema?.type === 'string') {
+          if (payloadJsonSchema.schema?.format === 'binary') {
+            conversions.push(processFile(undefined, params.body));
           }
-        } else if (file instanceof stream.Readable) {
-          conversions.push(
-            new Promise(resolve => {
-              resolve(getStream.buffer(file));
-            }).then(buffer => {
-              // This logic was taken from the `datauri` package, and ideally it should be able to accept the content
-              // of a file, or a file stream, but I'll PR that later to that package.
-              // @todo
-              const base64 = buffer.toString('base64');
-              const mimeType = mimer(file.path);
-              const filepath = path.basename(file.path);
-
-              params.body[prop] = `data:${mimeType};name=${encodeURIComponent(filepath)};base64,${base64 || ''}`;
-
-              return Promise.resolve(true);
-            })
-          );
         }
-      });
 
-    await Promise.all(conversions);
+        await Promise.all(conversions)
+          .then(fileMetadata => fileMetadata.filter(Boolean))
+          .then(fm => {
+            fm.forEach(fileMetadata => {
+              if (!fileMetadata) {
+                // If we don't have any metadata here it's because the file we have is likely
+                // the full string content of the file so since we don't have any filenames to
+                // work with we shouldn't do any additional handling to the `body` or `files`
+                // parameters.
+                return;
+              }
+
+              if (fileMetadata.paramName) {
+                params.body[fileMetadata.paramName] = fileMetadata.base64;
+              } else {
+                params.body = fileMetadata.base64;
+              }
+
+              params.files[fileMetadata.filename] = fileMetadata.buffer;
```

Comment:
With this we'll now be sending file buffers into `fetch-har` to use with the fetch request (instead of relying on the base64).